### PR TITLE
Asks before uploading the file to imgur

### PR DESCRIPTION
**Pull request by hari-govind**
Addresses [Give option to upload image](https://github.com/dibyadas/imagery/issues/3) issue.

### CODE REVIEW

**Comment by dibyadas** [comment]
@ashishkg0022  Yup

File: app.py [hunk 1]
```diff
@@ -91,7 +158,8 @@ def hello():
 			if file_data['file']['size']/(1024**2) > 20:
 				raise Exception("File too large (> 20MB)")
 			file_permalink = file_data['file']['url_private_download']
-			i = pool.apply_async(download_file, [file_permalink, file_id, channel_id, comment, user_id])
+			i = pool.apply_async(send_ephemeral, [user_id,channel_id,file_permalink,file_id,comment])
```
Comment (ashishkg0022) [inline app.py]: put spaces after commas.
Rest all looks good to me.

File: app.py [hunk 2]
```diff
@@ -55,15 +56,81 @@ def delete_link(user_id, channel_id, ts):
...
+
+
+@app.route('/app', methods=['GET','POST'])
 def hello():
 	json_data = request.json
 	try:
 		challenge = json_data['challenge']
 		return challenge
-	except KeyError:
+	except Exception as e:
```
Comment (dibyadas) [inline app.py]: I think the exceptions should be semantically separated. Here, we know that a if a KeyError has occurred while searching for `challenge` that means, it's not a verification request but a normal imagery request and the KeyError is not actually an error. Hence, keep the KeyError as it is and add another Exception block to catch any unexpected exceptions.

File: app.py (hunk 1 above)
Comment (dibyadas) [reply app.py]: @hari-govind for now, fix this, and later I would hope someone to PEP8-ify the entire codebase.

**Comment by dibyadas** [comment]
Closes #3

**Comment by vibhavagarwal5** [comment]
@hari-govind Can u document your functions for a better understanding for the future contributors.

**Comment by vibhavagarwal5** [comment]
Also @hari-govind it would be great if you could squash similar commits like 'fix typo' and etc together so as to make more meaningful commits for just 1 feature addition. Great work but 👍.

**Comment by hari-govind** [comment]
@vibhavagarwal5 How to do that bro ?

**Comment by vibhavagarwal5** [comment]
Check [this](https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit) @hari-govind

**Comment by dibyadas** [comment]
Awesome work @hari-govind ! Thanks!

**Pull request merged**